src/tests/conformance/messages.rs

```diff
@@ -184,6 +184,129 @@ async fn ignores_unsolicited_responses() {
     node.stop().await;
 }
 
+#[tokio::test]
+async fn eagerly_crawls_network_for_peers() {
+    // ZG-CONFORMANCE-012
+    //
+    // The node crawls the network for new peers and eagerly connects.
+    //
+    // Test procedure:
+    //
+    //  1. Create a set of peer nodes, listening concurrently
+    //  2. Connect to node with another main peer node
+    //  3. Wait for `GetAddr`
+    //  4. Send set of peer listener node addresses
+    //  5. Expect the node to connect to each peer in the set
+    //
+    // This test currently fails for zcashd; zebra passes (with a caveat).
+    //
+    // Current behaviour:
+    //
+    //  zcashd: Has different behaviour depending on connection direction.
+    //          If we initiate the main conneciton it sends Ping, GetHeaders,
+    //          but never GetAddr.
+    //          If the node initiates then it does send GetAddr, but it never connects
+    //          to the peers.
+    //
+    // zebra:   Passes with flying colors, so long as we keep responding on the main connection.
+    //          If we do not keep responding, then the peer connections take really long to establish,
+    //          sometimes even spuriously failing the test completely.
+
+    let (zig, node_meta) = read_config_file();
+
+    // create tcp listeners for peer set (port is only assigned on tcp bind)
+    let mut listeners = Vec::new();
+    for _ in 0u8..5 {
+        listeners.push(TcpListener::bind(zig.new_local_addr()).await.unwrap());
+    }
+
+    // get list of peer addresses
+    let peer_addresses = listeners
+        .iter()
+        .map(|listener| listener.local_addr().unwrap())
+        .collect::<Vec<_>>();
+
+    // start the node
+    let mut node = Node::new(node_meta);
+    node.start_waits_for_connection(zig.new_local_addr())
+        // .log_to_stdout(true)
+        .start()
+        .await;
+
+    // start peer listeners which "pass" once they've accepted a connection, and
+    // "fail" if the timeout expires. Timeout must be quite long, seems to take around
+    // 20-60 seconds for zebra.
+    let mut peer_handles = Vec::with_capacity(listeners.len());
+    for listener in listeners {
+        peer_handles.push(tokio::time::timeout(
+            tokio::time::Duration::from_secs(120),
+            tokio::spawn(async move {
+                println!(
+                    "{:?}: waiting for connection",
+                    listener.local_addr().unwrap()
+                );
+                listener.accept().await.unwrap();
```

Comment:
We may want to check the handshake completes here? 
```suggestion
                respond_to_handshake(listener).await.unwrap();
```

Reply:
Sure we can do that.